src/__tests__/fixtures/regression.css

```diff
@@ -0,0 +1,7 @@
+@custom-selector :--button .button;
+
+:--button {
+  &--active {
```

Comment:
@MoOx and that's fine but the spec need to take in consideration the `de-sugaring` of the custom selector. That's why I think we need some feedback from @tabatkins about that. 

I understand that use `matches` but it should take in consideration the `custom selectors`